### PR TITLE
Implemented the BinaryFileOutputTask class

### LINKED ISSUE #220: Missing BinaryFileOutputTask class of the main.task subpackage
**Actual Behaviour.**  
The BinaryFileOutputTask class is missing in the Android version of the application.

**Expected Behaviour**  
The BinaryFileOutputTask class should be implemented in the Android app for the completion of the porting of main.task subpackage.

**Would you like to work on the issue?**
I would like to work on this issue.

### PR DESCRIPTION
**Pull request by suresh-gandhi**
Fixes #220.

**Changes**: 
* Implemented the BinaryFileOutputTask class (belonging to the main subpackage) in Android version.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications are done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: [Compress the `app-debug.apk` file into a `<feature>.rar` or `<feature>.zip` file and upload it here]

### CODE REVIEW

**Comment by suresh-gandhi** [comment]
@jddeep Could you give me a peer review.

File: app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    public void setWrite(boolean write) {
+        this.write = write;
+    }
+
+    @Override
+    public void run() {
+        if ((write) && (!nfbServer.getInputData().isEmpty())) {
+            try {
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:52]: @suresh-gandhi Motive of changes is good. Use try with resources.

File: app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Override
+    public void init() {
+        try {
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:77]: Same. Try with resources

File: app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                out.close();
+            out = new FileOutputStream(fileName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public void stop() {
+        try {
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:88]: Use try with resources.

**Review by jddeep — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java (hunk 1 above)
Comment (suresh-gandhi) [reply app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:52]: out is a global variable used in a couple of more functions init and stop (which are necessary to implement being a part and parcel of the Task Interface) and hence we would be requiring an update to Java 9 to use try with resources here or other option would be to use finally block.

Comment (jddeep) [reply app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:52]: Well, I have implemented try with resources using Java 8.

Comment (suresh-gandhi) [reply app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:52]: @jdeep you can't do it with out-of-scope variables in Java-8. You must have done with in-scope variables.

Comment (suresh-gandhi) [reply app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:52]: @jddeep I am adding the finally block and updating the PR as I believe this is the best we can do, currently.

Comment (jddeep) [reply app/src/main/java/io/neurolab/main/task/BinaryFileOutputTask.java:52]: @suresh-gandhi Alright. Yes you can add the finally block here.

**Comment by suresh-gandhi** [comment]
@mariobehling @CloudyPadmal peer review done. Could it be merged now?

**Review by mariobehling — APPROVED** [review]

**Pull request merged**